libs/api/domains/auth-admin/src/lib/client/clients.service.ts

```diff
@@ -313,30 +312,21 @@ export class ClientsService extends MultiEnvironmentService {
     })
   }
 
-  async deleteClient(
-    user: User,
-    input: DeleteClientInput,
-  ): Promise<DeleteClientResponse[]> {
-    const response = input.environments.map((environment) => ({
-      environment,
-      success: true,
-    }))
-
+  async deleteClient(user: User, input: DeleteClientInput): Promise<boolean> {
     await Promise.all(
-      response.map((resp, index) =>
-        this.adminApiByEnvironmentWithAuth(resp.environment, user)
+      Object.values(Environment).map((resp) =>
+        this.adminApiByEnvironmentWithAuth(resp, user)
           ?.meClientsControllerDelete({
             tenantId: input.tenantId,
             clientId: input.clientId,
           })
           .catch((error) => {
-            response[index].success = false
-            this.handleError(error, resp.environment)
+            this.handleError(error, resp)
```

Comment:
The `handleError` swallows errors, so this will cause the service to always return true, even though none of the request succeeds. So the service needs to keep track of success of each request, and then check if one of it succeeds then return true otherwise return false.